client/lib/url/index.js

```diff
@@ -107,6 +108,41 @@ function urlToSlug( url ) {
 	return withoutHttp( url ).replace( /\//g, '::' );
 }
 
+/**
+ * Checks if the supplied string appears to be a URL.
+ * Looks only for the absolute basics:
+ *  - does it have a .suffix?
+ *  - does it have at least two parts separated by a dot?
+ *
+ * @param  {String}  query The string to check
+ * @return {Boolean} Does it appear to be a URL?
+ */
+function resemblesUrl( query ) {
+	let parsedUrl = url.parse( query );
+
+	// Make sure the query has a protocol - hostname ends up blank otherwise
+	if ( ! parsedUrl.protocol ) {
+		parsedUrl = url.parse( 'http://' + query );
```

Comment:
That's a cool idea. I won't do it in this particular spot (`resemblesUrl`) because it might be reused by other things, but worth us keeping in mind.